pay-api/src/pay_api/models/payment.py

```diff
@@ -34,22 +35,27 @@
 from .payment_status_code import PaymentStatusCode
 from .payment_system import PaymentSystem
 from .payment_transaction import PaymentTransactionSchema
-from .base_schema import BaseSchema
+from .invoice_reference import InvoiceReference
 
 
-class Payment(Audit):  # pylint: disable=too-many-instance-attributes
+class Payment(BaseModel):  # pylint: disable=too-many-instance-attributes
     """This class manages all of the base data about Payment ."""
 
     __tablename__ = 'payment'
 
     id = db.Column(db.Integer, primary_key=True, autoincrement=True)
     payment_system_code = db.Column(db.String(10), ForeignKey('payment_system.code'), nullable=False)
+    payment_account_id = db.Column(db.Integer, ForeignKey('payment_account.id'), nullable=True)
     payment_method_code = db.Column(db.String(15), ForeignKey('payment_method.code'), nullable=False)
-    payment_status_code = db.Column(db.String(20), ForeignKey('payment_status_code.code'), nullable=False)
+    payment_status_code = db.Column(db.String(20), ForeignKey('payment_status_code.code'), nullable=True)
+    invoice_number = db.Column(db.String(50), nullable=True, index=True)
```

Comment:
so we keep the invoice_number against payment? I thought its gonna be invoice_ref number..i mean link to invoice reference table
looks like we store invoice_number is in both tables..so comment may not apply

Reply:
Invoice reference is a reference number from paybc, which is a constant. Initially I named as refernce number, then changed to invoice_number

Reply:
yeah..just the naming i figured out